parallel/bouncing_network/bouncing_network.pde

```diff
@@ -1,144 +1,191 @@
+    // 'remove List' which we clear from the bag after
+    // the iteration process is done.
+    void update() {
+        ArrayList<Ball> clearList = new ArrayList<Ball>();
+
+        for (Ball ball : balls) {
+            if (ball.update()) {
+                clearList.add(ball);
+            }
+        }
+
+        balls.removeAll(clearList);
+    }
+
+    void draw() {
+        for (Ball ball : balls) {
+            ball.draw();
         }
-      } else if ( b.ret(0) > width-radius ) {
-        if( rightIP != "" ) {
-          println("sending right");
-          right.write(b.toStr());
-          balls.remove(i);
+    }
+
+    void add(float x, float y, float xv, float yv, float radius) {
+        add(new Ball(x, y, xv, yv, radius));
+    }
+
+    void add(Ball ball) {
+        balls.add(ball);
+    }
+}
+
+class Ball {
+    float x;
+    float y;
+    float xv;
+    float yv;
+    float radius;
+
+    Ball(float x, float y, float xv, float yv, float radius) {
+        this.x = x;
+        this.y = y;
+        this.xv = xv;
+        this.yv = yv;
+        this.radius = radius;
+    }
+
+    // It would be cool if the balls sped up when they were
+    // over a certain area.
+    boolean update() {
+        x += xv;
+        y += yv;
+
+        if (x < radius) {
+            return goTo(left);
+        } else if (x > width - radius) {
+            return goTo(right);
+        } else if (y > height - radius || y < radius) {
+            flipY();
+        }
+
+        // Think about adding a new else if to check whether balls have collided
+
+        return false;
+    }
+
+    boolean goTo(Computer computer) {
+        if (computer.isConnected()) {
+            computer.send(this);
+            return true;
         } else {
-          b.flop(2);
-          b.update();
+            flipX();
+            return false;
         }
-      } else if ( b.ret(1) > height-radius || b.ret(1) < radius ) {
-        b.flop(3);
-        b.update();
-      } else { b.update(); }
-    }
-    // Think about adding a new else if to check whether balls have collided
-  }
-  void draw() {
-    for(i=0; i<balls.size(); i++) {
-      ball b = balls.get(i);
-      b.draw();
-    }
-  }
-  void addBall(float x, float y, float xv, float yv) {
-    balls.add(new ball(x, y, xv, yv));
-  }
+    }
+
+    // nf turns floats into strings
+    // join turns arrays of strings into strings
+    String toString() {
+        float[] points = {x, y, xv, yv, radius};
+
+        return join(nf(points, 0, 0), ",");
+    }
+
+    void draw() {
+        ellipse(x, y, radius * 2, radius * 2);
+    }
+
+    void flipX() {
+        xv *= -1.0;
+    }
+
+    void flipY() {
+        yv *= -1.0;
+    }
 }
 
+class Computer {
+    String ip;
+    int port;
+    Client conn;
+
+    Computer(String ip, int port) {
+        this.ip = ip;
+        this.port = port;
+        reconnect();
+    }
+
+    void reconnect() {
+        conn = new Client(self, ip, port);
```

Comment:
I believe this was the purpose behind holding a global reference to the current applet instance.